### PR TITLE
KHP3-7165 - feat/Print item transactions (Bin Card and Stock Card)

### PR DESCRIPTION
**Pull request by PatrickWaweru**
Added the ability to print a BIN CARD using the item transactions.

## Requirements

- [X] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [X] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [X] My work includes tests or is validated by existing tests.

#### New Print Buttons
items transactiosn 2024-12-13 09-19-33
#### Actual BIN Card Printout
BIN Card Printout 2024-12-13 12-54-39

### CODE REVIEW

**Comment by ojwanganto** [comment]
Please create a ticket for this work. Also, share a sample output for the print

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import styles from './printable-transaction-footer.scss';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import { useSession } from '@openmrs/esm-framework';
+
+type PrintableFooterProps = {
+  // facilityInfo: Record<string, any>;
```
Comment (donaldkibet) [inline src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx:8]:
```suggestion
 
```

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <div className={styles.container}>
+      <p className={styles.itemFooter}>{title}</p>
+      <p className={styles.footDescription}>
+        {t(
+          'generatedMessage',
+          'The card has been electronically generated and is a valid document. It was created by {{userName}} on {{date}} at {{time}}',
+          {
+            userName: `${session?.user?.display}`,
+            date: dayjs().format('DD-MM-YYYY'),
```
Comment (donaldkibet) [inline src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx:25]: we have a function formatDate from `@openmrs/esm-framework` to do the date formating.

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-header.component.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  return (
+    <div className={styles.container}>
+      <div className={styles.printableHeader}>
+        <p className={styles.heading}>{t('bincard', 'Bin Card')}</p>
+        {logo?.src ? (
+          <img className={styles.img} height={60} width={250} src={logo.src} alt={logo.alt} />
+        ) : logo?.name ? (
+          logo.name
+        ) : (
+          // OpenMRS Logo
```
Comment (donaldkibet) [inline src/stock-items/add-stock-item/transactions/printout/printable-transaction-header.component.tsx:24]:
```tsx
   <svg role="img" className={styles.logo}>
      <title>{t('openmrsLogo', 'OpenMRS logo')}</title>
      <use href="#omrs-logo-full-color"></use>
    </svg> 
```

Comment (donaldkibet) [reply src/stock-items/add-stock-item/transactions/printout/printable-transaction-header.component.tsx:24]: We can use this instead duplicating the svg contents

**Review by donaldkibet — APPROVED** [review]
Thanks @PatrickWaweru left some suggestions

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx (hunk 1 above)
Comment (PatrickWaweru) [reply src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx:8]: done

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-header.component.tsx (hunk 3 above)
Comment (PatrickWaweru) [reply src/stock-items/add-stock-item/transactions/printout/printable-transaction-header.component.tsx:24]: This method works when displaying the preview component but it does not print out when printed

File: src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx (hunk 2 above)
Comment (PatrickWaweru) [reply src/stock-items/add-stock-item/transactions/printout/printable-transaction-footer.component.tsx]: done

**Comment by ojwanganto** [comment]
Please include the facility name in the header. Potentially, drop the OpenMRS logo for now until we are ready to configure the facility logo

**Review by makombe — APPROVED** [review]
Thanks @PatrickWaweru

**Pull request merged**